src/content/docs/paper/dev/api/menu-type-api.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Menu Type API
+slug: paper/dev/menu-type-api
+description: A guide to the Menu Type API.
+version: 1.21.8
+---
+
+Minecraft has a lot of types of menus. From chests, to crafting tables, to anvils, and even villager trade menus.
+With the old Bukkit inventory API, it was not possible to replicate most of these perfectly. Exactly for
+this reason, the menu type API was introduced.
+
+## What is a menu?
+Menus, also referred to as views, are user interfaces, which can be created and viewed by players. The
+[`MenuType`](jd:paper:org.bukkit.inventory.MenuType) interface declares all possible menu types a menu can have.
+The difference between menus and inventories is that inventories are containers and menus are
+their visual representations.
+
+Menus created using the menu type API **follow the same logic as Vanilla**, meaning they are fully
+functional. Some of them can directly represent a block (a.e. a furnace) and the
+[`MERCHANT`](jd:paper:org.bukkit.inventory.MenuType#MERCHANT) type can represent a merchant entity in the world.
+
+## What are inventory views?
+An [`InventoryView`](jd:paper:org.bukkit.inventory.InventoryView) is a specific view created from a menu type.
+In the general sense, an inventory view links together **two separate inventories** and always has a player viewing them.
+The bottom linked inventory is the player's inventory.
+
+There are specialized subinterfaces for getting specific inventory views (a.e. [`FurnaceView`](jd:paper:org.bukkit.inventory.view.FurnaceView)
+for furnace inventories). This makes it very convenient to check whether an inventory view is of a specific type.
```

Comment:
That's fair tho all the view don't have their own sub type, so might mention InventoryView#getMenuType for more control depending on the use case.